processing/src/main/java/org/apache/druid/segment/AutoTypeColumnIndexer.java

```diff
@@ -282,6 +282,10 @@ public ColumnCapabilities getColumnCapabilities()
 
   private ColumnType getLogicalType()
   {
+    if (fieldIndexers.isEmpty()) {
+      // we didn't see anything, so we can be anything, so why not a string?
+      return ColumnType.STRING;
```

Comment:
I think it would be useful since null really is like its own thing, but i think the change has some implications since we probably also want a `ValueType.NULL` to go along with it and then a lot of places to handle the case.

Ive been considering this for quite a while now because it would make a lot of things a lot nicer, particularly with non-vectorized expressions where we have to treat `null` as a `STRING` unless something externally told us to treat it as some other type, so there are lots of strange places where we allow string type to participate in things which might otherwise be invalid because we don't have a null type.